src/client/components/pages/parts/identifier-types-tree.tsx

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (C) 2023 Shivam Awasthi
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.	See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+ */
+import React, {useCallback, useState} from 'react';
+import {Button} from 'react-bootstrap';
+import {IdentifierTypeDataT} from '../../forms/type-editor/typeUtils';
+import {orderBy as _orderBy} from 'lodash';
+
+
+type IdentifierTypeTreePropsT = {
+	identifierTypes: IdentifierTypeDataT[],
+	parentId?: number | null,
+	indentLevel?: number
+};
+
+function IdentifierTypeTree({identifierTypes, parentId, indentLevel}: IdentifierTypeTreePropsT) {
+	const [expandedIdentifierTypeIds, setExpandedIdentifierTypeIds] = useState([]);
+
+	function toggleExpand(identifierTypeId) {
+		setExpandedIdentifierTypeIds((prevExpandedIds) => {
+			if (prevExpandedIds.includes(identifierTypeId)) {
+				return prevExpandedIds.filter((id) => id !== identifierTypeId);
+			}
+			return [...prevExpandedIds, identifierTypeId];
+		});
+	}
+
+	const handleClick = useCallback((event) => {
+		const identifierTypeId = parseInt(event.target.value, 10);
+		toggleExpand(identifierTypeId);
```

Comment:
`event.target.valueAsNumber` returns undefined in this case. It seems to work in the previous case, but here I think its not working because we are passing `idenType.id` as a value prop in the Button, and then using it.